Update calibration-jobs.mdx to reflect updates in calibration strategy (#4180)
Qiskit/documentation: docs/guides/calibration-jobs.mdx

---
title: Monitoring, calibrations, and benchmarking
description: Describes what calibration jobs are and how often they run
---

# Monitoring, calibrations, and benchmarking

Calibrating a quantum computer requires optimizing a plethora of parameters defining the electrical signals that actuate the quantum gates and readout operations. Before a quantum computer is released, its initial calibration process involves thoroughly tuning each parameter to achieve the best possible performance based on benchmarks relevant to the quantum computer's expected workloads. Once a quantum computer has been released, the primary goal is to maintain consistent performance for the lifetime of the device. The optimal values of many of the calibrated parameters remain stable indefinitely, but some vary over time as a result of factors such as changes in the environment of two-level systems (TLS) on the quantum processor chip, changes in the ambient conditions (for example, temperature) in the data center, or instability within the control systems.

> To ensure consistent performance, IBM&reg; quantum computers frequently monitor parameters that may drift over time, run calibrations when needed, and perform daily benchmarking. This page details three processes — monitoring, calibrating, and benchmarking — that work together to ensure that the IBM fleet of quantum computers remains as stable, predictable, and available to users as possible.

Reviewer: Checking the language of "IBM quantum computers frequently monitor".... do the computers themselves monitor (they run the jobs automatically)?

Reviewer 2: The jobs run automatically; is that "part" of the quantum computer, not sure. It's part of the "system" though.

### Single-qubit operations

These calibrations ensure accurate implementation of the single-qubit gates: sx, x, rx (fractional). We adjust:

* Qubit frequencies
* Pulse amplitudes and phases

These calibrations are batched across the affected qubits and executed concurrently where appropriate, with batching strategies tailored to each calibration type.

### Two-qubit operations

* CZ and RZZ gate amplitudes and phases (for Heron and Nighthawk processors)
* ECR gate amplitudes and phases (for Eagle processors)

These calibrations are run in batches of non-nearest-neighbor qubits to minimize crosstalk.

### Readout

* Readout pulse angles
* Measurement discrimination parameters

These calibrations are run simultaneously on qubits that require calibration.

### How calibrations are scheduled

* A calibration job cannot run simultaneously while an external job or session is running.
* Therefore, during long sessions, the quantum computer may experience reduced effective stability due to delayed or infrequent recalibration.
* Two jobs submitted at the same time might run under different calibration sets, depending on timing.

> * All circuits within a single job always use one consistent calibration.

Reviewer: Is the solution to remove this line, or to edit to `Though there is no guarantee, all circuits within a single job nearly always use one consistent calibration.`?

Multiple aspects of the QPU are affected by the many different calibrations that IBM&reg; runs daily and hourly.

### Device calibrations

> Calibration for qubit and gate parameters are conducted on an as-needed basis when parameters are determined to require calibration based on regularly scheduled diagnostic monitoring experiments that are interleaved with user jobs. These calibrations can occur at any time during the day based on device performance. Experiments which measure benchmarks which are reported to Qiskit are run daily and interleave with user jobs.

Reviewer: ```suggestion
Calibrations for qubit and gate parameters are conducted on an as-needed basis. Regularly scheduled diagnostic monitoring experiments, interleaved with user jobs, determine when calibrations are needed. These calibrations can occur at any time during the day, based on device performance. Benchmark-measuring experiments that are reported to Qiskit are run daily and interleave with user jobs.
```

*   **Single-qubit calibrations**

    The frequency, pulse amplitudes, and “DRAG” parameter of each qubit are calibrated on an as-needed basis.

*   **Single-qubit benchmarking**

    The T1/T2 coherence times and measurement error rates are measured, and randomized benchmarking is run in batches of non-adjacent qubits. These benchmarks are measured on a daily basis, with the exception of measurement error, which is reported when readout angle calibrations are performed as well. 

*   **Two-qubit gate calibrations and benchmarking**

>     The amplitude and phase of each pulse is calibrated, and randomized benchmarking in batches of well-separated gates of similar length is run. These benchmarks are measured on a daily basis.

Reviewer 2: This part is a little confusing. Are the two-qubit gates calibrated as needed as well? Add that layer fidelity is also measured daily.

> ### Hourly calibrations/monitoring

Reviewer 2: I would change this to just "Hourly monitoring" and put the readout angle calibration into the single qubit section.

It's also confusing what we mean about stability checks vs system health? And when we say some calibrations are "as-needed" can we elaborate on how these system checks trigger the calibrations?

*   **Single-qubit calibrations**

    The frequency, pulse amplitudes, and “DRAG” parameter of each qubit are calibrated on an as-needed basis.

*   **Single-qubit benchmarking**

>     The T1/T2 coherence times and measurement error rates are measured, and randomized benchmarking is run in batches of non-adjacent qubits. These benchmarks are measured on a daily basis, other than measurement error, which is reported when readout angle calibrations are performed as well. 

Reviewer: ```suggestion
    The T1/T2 coherence times and measurement error rates are measured, and randomized benchmarking is run in batches of non-adjacent qubits. These benchmarks are measured on a daily basis, with the exception of measurement error, which is reported when readout angle calibrations are performed as well. 
```
